### ISSUE TITLE
[BUG]: `create_entity_permutations` occasionally segfaults for prism meshes

### ISSUE THREAD

**Issue by jpdean** [FEniCS/dolfinx#3208]
### Summarize the issue

Calling `create_entity_permutations` occasionally segfaults for prism meshes. For some reason, I can only reproduce this on a clean Ubuntu install, it seems to run without failing on the docker images and on my local install.

### How to reproduce the bug

Run the minimal example below on a minimal Ubuntu install as described in the build and test workflow [here](https://github.com/FEniCS/dolfinx/blob/main/.github/workflows/ccpp.yml#L84).

### Minimal Example (Python)

```Python
from mpi4py import MPI
from dolfinx import mesh

n = 3
msh = mesh.create_unit_cube(MPI.COMM_WORLD, n, n, n, cell_type=mesh.CellType.prism)
msh.topology.create_entity_permutations()
```

### Version

main branch

### Installation

Follow the build and run workflow file [here](https://github.com/FEniCS/dolfinx/blob/main/.github/workflows/ccpp.yml#L84) using a plain Ubuntu image.

**Comment by chrisrichardson** [comment]
I can reproduce this bug.
I think it stems from `compute_face_permutations` in `permutationcomputation.cpp` where there are separate computations for simplices and 'tp' elements. However, I don't think `prism` is being properly treated.

**Comment by chrisrichardson** [comment]
Yes, confirmed: using `auto f_to_v = topology.connectivity(2, 0);` is not enough for prism cells, now they contain two types of facet, we need to use `auto f0_to_v = topology.connectivity({2, 0}, {0, 0});` and `auto f1_to_v = topology.connectivity({2, 1}, {0, 0});`
Probably this function (`compute_face_permutations`) needs general refactoring.

**Comment by chrisrichardson** [comment]
Fixed by  #3223
